assets/RNAseq_report.Rmd

```diff
@@ -530,27 +1661,29 @@ Hover over the dots to reveal the pathway names.
 The table below provides more detail on all enriched pathways."))
 ```
 
-```{r, eval=params$pathway_analysis, out.height = "100%", out.width = "100%", warning=FALSE, message=FALSE}
-    library(gprofiler2)
-    #library(rlist)
-    #TODO: check if here, only the filtered genes are necessary
+```{r gost, eval=params$pathway_analysis, out.height = "100%", out.width = "100%", warning=FALSE, message=FALSE}
     contrast_files <- list.files(path=paste0("differential_gene_expression/DE_genes_tables/"))
     q_list <- list()
     q_names <- c()
+
     #Use non-allgenes files for PA to only show DE genes
     for (file in contrast_files){
+
         #Reading DE genes list
         fname <- tools::file_path_sans_ext(basename(file))
         DE_genes <- read.csv(file = paste0("differential_gene_expression/DE_genes_tables/", file), sep="\t", header = T)
         if(nrow(DE_genes) <= 1){
-        next
+            next
         }
         q = as.character(DE_genes$Ensembl_ID)
         q_list <- append(q_list, list(q))
         q_names <- append(q_names, fname)
     }
     datasources <- c("KEGG","REAC")
     names(q_list) <- q_names
+    if (!length(q_list)) {
+        stop("list empty?!?")
```

Comment:
Could be a bit more specific, which list? What does the user need to check?

Reply:
Oops, that is a left-over from some debugging!